### PR TITLE
ddl, partition: make partition table can use binary type column 

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/23464

Problem Summary:

### What is changed and how it works?

make partition table can use binary type column 

### Check List 

Tests 

- Unit test

### Release note 

- make partition table can use binary type column

### CODE REVIEW

**Comment by blacktear23** [comment]
@xiongjiwei what about add some unit-test for add partitions.

File: ddl/db_partition_test.go [hunk 1]
```diff
@@ -583,6 +583,9 @@ create table log_message_1 (
 
 	tk.MustExec("drop table if exists t;")
 	tk.MustExec(`create table t(a int) partition by range (a) (partition p0 values less than (18446744073709551615));`)
+
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec(`create table t(a binary) partition by range columns (a) (partition p0 values less than (X'0C'));`)
```
Comment (djshow832) [inline ddl/db_partition_test.go:588]: Add test cases to insert some values.

**Pull request merged**